File: Data Structures & Algorithms/concatenation-of-array/submission-0.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int numSize = nums.size();

        std::vector<int> ans(2*numSize,0);

        int k=0;
        for(int i=0;i<ans.size();i++)
        {
            ans[i]=nums[k];
            k++;
            if(k==numSize)
                k=0;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        n = len(nums)
        pref = [1] * n
        sufx = [1] * n

        p = 1
        s = 1
        for i in range(1, n):
            pref[i] = p * nums[i - 1]
            p = pref[i]

            sufx[n - i - 1] = s * nums[n - i]
            s = sufx[n - i - 1]

        for i in range(n):
            pref[i] = pref[i] * sufx[i]
        return pref
